servicenow: Add option to parse fields containing only display values in S3.

kcreddy opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

## Proposed commit message
To avoid mapping failures, https://github.com/elastic/integrations/pull/12018 added support for diverse inputs. But when the users store only the display values inside the S3 bucket, it leads to failures in multiple processors. This is because the ingest pipeline converts scalar fields into objects with `.value` and we currently have ~150 `convert` processors on fields containing `.value` which all fail because users may not have intended to use them in `.value` form. 

Providing an option to the users to convert into an object with `.display_value` will remove these processor failures just like the users intended to store their data in S3.

Other changes:
- Updated README describing the option inside Troubleshooting section.
- Added new log file for pipeline testing with new option.
- A dashboard format is auto-generated by `elastic-package`.

<!-- Mandatory
Explain here the changes you made on the PR.

Please explain:

- WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
- WHY:  the rationale/motivation for the changes

This text will be pasted into the squash dialog when the change is committed and will be
a long term historical record of the change to help future contributors understand the
change, please help them by making it clear and comprehensive, they may be you.

If the commit title is adequate to describe both of these things, The text here may be omitted
or replaced with "See title". The title of the PR will be used as the commit message title when
the merge is made and the "See title" marker will be removed if present.

The text here and the PR title will be subject to the PR review process.
-->

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 
## How to test this PR locally

<!-- Recommended
Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->
Added sample logs in the file: `test-event-aws-with-display-values.log` to work with this change.

Run pipeline tests:
Command: `cd packages/servicenow && elastic-package stack down && elastic-package build && elastic-package stack up --version=8.16.2 -d -v  && eval "$(elastic-package stack shellinit)" && elastic-package test pipeline --generate -v`
Result:
```
--- Test results for package: servicenow - START ---
╭────────────┬─────────────┬───────────┬───────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME                                                         │ RESULT │ TIME ELAPSED │
├────────────┼─────────────┼───────────┼───────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ servicenow │ event       │ pipeline  │ (ingest pipeline warnings test-event-aws-with-display-values.log) │ PASS   │ 359.851458ms │
│ servicenow │ event       │ pipeline  │ (ingest pipeline warnings test-event-aws.log)                     │ PASS   │  367.12125ms │
│ servicenow │ event       │ pipeline  │ (ingest pipeline warnings test-event.log)                         │ PASS   │  365.30875ms │
│ servicenow │ event       │ pipeline  │ test-event-aws-with-display-values.log                            │ PASS   │ 117.846334ms │
│ servicenow │ event       │ pipeline  │ test-event-aws.log                                                │ PASS   │     82.298ms │
│ servicenow │ event       │ pipeline  │ test-event.log                                                    │ PASS   │ 1.031382583s │
╰────────────┴─────────────┴───────────┴───────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: servicenow - END   ---
Done
```

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Supersedes #123
-->
- 

## Screenshots

<!-- Optional
Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration
- dashboards with collected metrics or logs
-->
<img width="1091" alt="Screenshot 2025-01-16 at 12 09 05 AM" src="https://github.com/user-attachments/assets/af81d342-3ad1-42df-98e9-dbddc6ee0749" />

<img width="298" alt="Screenshot 2025-01-16 at 11 35 21 AM" src="https://github.com/user-attachments/assets/2ffa9516-d547-46a8-b715-1e4ff66b60fb" />

elasticmachine commented:
Pinging @elastic/security-service-integrations (Team:Security-Service Integrations)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/20455)
* Commit: ac2c7c4de8e564ac2a5bcb2c2e36ef7b910f934a

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/20455","number":20455,"commit":"ac2c7c4de8e564ac2a5bcb2c2e36ef7b910f934a"}],"number":20455}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/20456)
* Commit: 0898c584bd7318d496c63c9a6186bb2bc6ca6cb9

### History
* :green_heart: [Build #20455](https://buildkite.com/elastic/integrations/builds/20455) succeeded ac2c7c4de8e564ac2a5bcb2c2e36ef7b910f934a

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/20456","number":20456,"commit":"0898c584bd7318d496c63c9a6186bb2bc6ca6cb9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/20455","number":20455,"commit":"ac2c7c4de8e564ac2a5bcb2c2e36ef7b910f934a"}],"number":20456}
buildkite-pr-comment-->

ShourieG reviewed (approved):
LGTM, nit only - for " Data Has Only Display Values", should we keep it in advanced setting by setting show_user:false ?

kcreddy commented:
> LGTM, nit only - for " Data Has Only Display Values", should we keep it in advanced setting by setting show_user:false ?

I am okay with that. Moved to `Advanced options` and updated README.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/20471)
* Commit: 8b328a0444c8c52f1b0d38e4e423ecef4db3ada9

### History
* :green_heart: [Build #20456](https://buildkite.com/elastic/integrations/builds/20456) succeeded 0898c584bd7318d496c63c9a6186bb2bc6ca6cb9
* :green_heart: [Build #20455](https://buildkite.com/elastic/integrations/builds/20455) succeeded ac2c7c4de8e564ac2a5bcb2c2e36ef7b910f934a

cc @kcreddy

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/20471","number":20471,"commit":"8b328a0444c8c52f1b0d38e4e423ecef4db3ada9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/20456","number":20456,"commit":"0898c584bd7318d496c63c9a6186bb2bc6ca6cb9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/20455","number":20455,"commit":"ac2c7c4de8e564ac2a5bcb2c2e36ef7b910f934a"}],"number":20471}
buildkite-pr-comment-->
